libresonic-main/src/main/java/org/libresonic/player/controller/LoginController.java

```diff
@@ -0,0 +1,72 @@
+package org.libresonic.player.controller;
+
+import org.libresonic.player.Logger;
+import org.libresonic.player.domain.User;
+import org.libresonic.player.service.PlaylistService;
+import org.libresonic.player.service.SecurityService;
+import org.libresonic.player.service.SettingsService;
+import org.libresonic.player.util.StringUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Spring MVC Controller that serves the login page.
+ */
+@Controller
+@RequestMapping("/login")
+public class LoginController {
+
+
+    private static final Logger LOG = Logger.getLogger(LoginController.class);
+
+    private final SecurityService securityService;
+    private final SettingsService settingsService;
+    private final PlaylistService playlistService;
+
+    @Autowired
+    public LoginController(SecurityService securityService, SettingsService settingsService, PlaylistService playlistService) {
```

Comment:
Right fixed it. Agree with the "standard" consideration. In a similar idea I wondered to change the signature of the login method as Spring supports a large range of solutions. But it would lead to significant code changes with regression risks.